### PR TITLE
bump(*): k8s 1.18.0, alpha-build-machinery -> build-machinery-go

### PR DESCRIPTION
**Pull request by gabemontero**
/assign @adambkaplan

### CODE REVIEW

**Comment by gabemontero** [comment]
https://github.com/openshift/release/pull/8321 up to switch build-utils to golang 1.13 so make build works with build-machinery-go

File: .idea/workspace.xml [hunk 1]
```diff
@@ -0,0 +1,45 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (tnozicka) [inline .idea/workspace.xml]: not this file
/hold

File: Makefile [hunk 2]
```diff
@@ -8,6 +8,6 @@ GO_BUILD_PACKAGES :=./pkg/...
 # You can list all the golang related variables by:
 #   $ make -n --print-data-base | grep ^GO
 
-include vendor/github.com/openshift/library-go/alpha-build-machinery/make/golang.mk
+include vendor/github.com/openshift/build-machinery-go/make/golang.mk
```
Comment (tnozicka) [inline Makefile:11]: these changes don't belong to bump commit

File: go.mod [hunk 3]
```diff
@@ -1,17 +1,18 @@
...
 
-go 1.12
+go 1.13
 
 require (
 	github.com/BurntSushi/toml v0.3.1
 	github.com/containers/image v3.0.2+incompatible
 	github.com/opencontainers/go-digest v1.0.0-rc1 // indirect
 	github.com/opencontainers/image-spec v1.0.1 // indirect
 	github.com/openshift/api v3.9.1-0.20191002160657-d92789481b05+incompatible
```
Comment (tnozicka) [inline go.mod]: this version needs fixing as per https://github.com/openshift/openshift-controller-manager/pull/85#discussion_r408247518

**Review by tnozicka — CHANGES_REQUESTED** [review]

File: .idea/workspace.xml (hunk 1 above)
Comment (gabemontero) [reply .idea/workspace.xml]: removed

Comment (gabemontero) [reply .idea/workspace.xml]: /hold cancel

File: Makefile (hunk 2 above)
Comment (gabemontero) [reply Makefile:11]: separated into separate commit

File: go.mod (hunk 3 above)
Comment (gabemontero) [reply go.mod]: version updated to matched to what we current have in https://github.com/openshift/openshift-controller-manager/pull/85

File: build/doc.go [hunk 4]
```diff
@@ -2,5 +2,5 @@ package build
 
 import (
 	// Import alpha-build-machinery to get standard makefile targets
-	_ "github.com/openshift/library-go/alpha-build-machinery"
+	_ "github.com/openshift/build-machinery-go"
```
Comment (adambkaplan) [inline build/doc.go:5]: for me, this is OK to go into the bump commit. iirc `go mod tidy` will clean up the `go.mod` file if it doesn't find a direct or indirect reference.

**Review by adambkaplan — APPROVED** [review]
/lgtm

File: build/doc.go (hunk 4 above)
Comment (tnozicka) [reply build/doc.go:5]: You could run `go mod tidy` and only split the files into correct commits after, or do the reverse order.

We are not enforcing this anywhere else yet, but there are strict requirements in origin to touch only glide files and vendor.

I think the best reason for splitting files correctly into bump commits is review. There is CI check named `verify deps` that makes sure `go mod tidy` was called and that all the files in vendor/ match what's written in `go.{mod,sum}`. With that the only files that need to be reviewed are `go.{mod,sum}`.

**Pull request merged**